src/main/java/org/openwebnet4j/message/Thermoregulation.java

```diff
@@ -0,0 +1,348 @@
+    }
+
+    public enum MODE {
+        HEATING(1),
+        CONDITIONING(2),
+        GENERIC(3);
+
+        private final Integer value;
+
+        private MODE(Integer value) {
+            this.value = value;
+        }
+
+        public static MODE fromValue(Integer i) {
+            Optional<MODE> m = Arrays.stream(values()).filter(val -> i.intValue() == val.value.intValue()).findFirst();
+            return m.orElse(null);
+        }
+    }
+
+    public enum LOCAL_OFFSET {
+        PLUS_3("03", "+3"),
+        PLUS_2("02", "+2"),
+        PLUS_1("01", "+1"),
+        NORMAL("00", "NORMAL"),
+        MINUS_1("11", "-1"),
+        MINUS_2("12", "-2"),
+        MINUS_3("13", "-3"),
+        OFF("4", "OFF"),
+        PROTECTION("5", "PROTECTION");
+
+        private final String value;
+        private final String label;
+
+        private LOCAL_OFFSET(String value, String label) {
+            this.value = value;
+            this.label = label;
+        }
+
+        public static LOCAL_OFFSET fromValue(String s) {
+            Optional<LOCAL_OFFSET> offset = Arrays.stream(values()).filter(val -> s.equals(val.value)).findFirst();
+            return offset.orElse(null);
+        }
+
+        public String getLabel() {
+            return label;
+        }
+    }
+
+    public enum DIM implements Dim {
+        TEMPERATURE(0),
+        TEMP_TARGET(12),
+        OFFSET(13),
+        TEMP_SETPOINT(14),
+        PROBE_TEMPERATURE(15),
+        VALVES_STATUS(19),
+        ACTUATOR_STATUS(20),
+        ACTUATOR_STATUS_ON(1),
+        ACTUATOR_STATUS_OFF(0);
+
+        private static Map<Integer, DIM> mapping;
+
+        private final int value;
+
+        private DIM(Integer value) {
+            this.value = value;
+        }
+
+        private static void initMapping() {
+            mapping = new HashMap<Integer, DIM>();
+            for (DIM d : values()) {
+                mapping.put(d.value, d);
+            }
+        }
+
+        public static DIM fromValue(int i) {
+            if (mapping == null) {
+                initMapping();
+            }
+            return mapping.get(i);
+        }
+
+        @Override
+        public Integer value() {
+            return value;
+        }
+    }
+
+    private static final int WHO = THERMOREGULATION.value();
+
+    protected Thermoregulation(String value) {
+        super(value);
+    }
+
+    @Override
+    protected What whatFromValue(int i) {
+        return WHAT.fromValue(i);
+    }
+
+    @Override
+    protected Dim dimFromValue(int i) {
+        return DIM.fromValue(i);
+    }
+
+    /**
+     * OpenWebNet message to Manual setting of “N” zone to T temperature <b>*#4*where*#14*T*M##</b>.
+     *
+     * @param where Zone between #1 and #99
+     * @param temp temperature T between 5.0° and 40.0° (with 0.5° step)
+     * @param mode
+     * @return message
+     */
+    public static Thermoregulation requestWriteSetpointTemperature(String where, float newSetPointTemperature,
```

Comment:
maybe add a check 5<=newSetPointTemperature<=40 and round to nearest 0.5°C step.